tests/ops/qubit/test_special_unitary.py

```diff
@@ -0,0 +1,388 @@
+        assert matrix.shape == (2**n, 2**n)
+        assert np.allclose(matrix @ matrix.conj().T, np.eye(2**n))
+
+    @pytest.mark.parametrize("n", [1, 2])
+    @pytest.mark.parametrize("seed", [214, 2491, 8623])
+    def test_special_unitary_matrix_random_broadcasted(self, n, seed):
+        """Test that ``special_unitary_matrix`` returns a correctly-shaped
+        unitary matrix for broadcasted random input parameters."""
+        np.random.seed(seed)
+        d = 4**n - 1
+        theta = np.random.random((2, d))
+        matrix = special_unitary_matrix(theta, n)
+        assert matrix.shape == (2, 2**n, 2**n)
+        assert all(np.allclose(m @ m.conj().T, np.eye(2**n)) for m in matrix)
+        separate_matrices = [special_unitary_matrix(t, n) for t in theta]
+        assert qml.math.allclose(separate_matrices, matrix)
+
+    @pytest.mark.parametrize("n", [1, 2, 3])
+    def test_special_unitary_matrix_single_param(self, n):
+        """Test that ``special_unitary_matrix`` returns a Pauli rotation matrix for
+        inputs with a single non-zero parameter, and that the parameter mapping
+        matches the lexicographical ordering of ``pauli_words``."""
+        d = 4**n - 1
+        words = pauli_words(n)
+        for word, theta in zip(words, np.eye(d)):
+            x = 0.2142
+            matrix = special_unitary_matrix(x * theta, n)
+            paulirot_matrix = qml.PauliRot(-2 * x, word, wires=list(range(n))).matrix()
+            assert np.allclose(matrix @ matrix.conj().T, np.eye(2**n))
+            assert np.allclose(paulirot_matrix, matrix)
+
+    @pytest.mark.parametrize("n, theta, expected", special_matrix_cases)
+    def test_special_unitary_matrix_special_cases(self, n, theta, expected):
+        """Tests that ``special_unitary_matrix`` returns the correct matrices
+        for a few specific cases."""
+        matrix = special_unitary_matrix(theta, n)
+        assert qml.math.allclose(matrix, expected)
+
+    def test_special_unitary_matrix_special_cases_broadcasted(self):
+        """Tests that ``special_unitary_matrix`` returns the correct matrices
+        for a few specific cases broadcasted together."""
+        _, thetas, exp = zip(*special_matrix_cases[1:])
+        theta = np.stack(thetas)
+        matrix = special_unitary_matrix(theta, 2)
+        assert qml.math.allclose(matrix, np.stack(exp))
+
+
+theta_1 = np.array([0.4, 0.1, 0.1])
+theta_2 = np.array([0.4, 0.1, 0.1, 0.6, 0.2, 0.3, 0.1, 0.2, 0, 0.2, 0.2, 0.2, 0.1, 0.5, 0.2])
+n_and_theta = [(1, theta_1), (2, theta_2)]
+
+
+class TestSpecialUnitary:
+    """Tests for the Operation ``SpecialUnitary``."""
+
+    @pytest.mark.parametrize("n, theta", n_and_theta)
+    def test_decomposition(self, n, theta):
+        """Test that SpecialUnitary falls back to QubitUnitary."""
+
+        wires = list(range(n))
+        decomp = qml.SpecialUnitary.compute_decomposition(theta, wires, n)
+        decomp2 = qml.SpecialUnitary(theta, wires).decomposition()
```

Comment:
do you have reason to believe these might be different, or is this just copying existing test standards? I see it in a few places here, as well as in existing code

Reply:
Yeah no, I just copied this behaviour from other tests. It somewhat ensures that `decomposition` is not overwritten separately, I suppose, for example by changing inheritance at some point. But I agree that it is a bit of an esoteric standard...